installation fails on missing clang name

**Bug summary**
```
cd /work/00434/eijkhout/adaptivecpp/build-23.10.0-milan-gcc13/src/runtime && /scratch/tacc/apps/gcc/13.2.0/bin/g++ -DHIPSYCL_DEBUG_LEVEL=2 -Dacpp_rt_EXPORTS -I/work/00434/eijkhout/adaptivecpp/adapti
vecpp-23.10.0/include -I/work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0 -I/work/00434/eijkhout/adaptivecpp/build-23.10.0-milan-gcc13/include -O3 -DNDEBUG -fPIC -std=c++17 -MD -MT src/runtime/CM
akeFiles/acpp-rt.dir/runtime.cpp.o -MF CMakeFiles/acpp-rt.dir/runtime.cpp.o.d -o CMakeFiles/acpp-rt.dir/runtime.cpp.o -c /work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0/src/runtime/runtime.cpp
In file included from /work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0/src/compiler/../../include/hipSYCL/compiler/FrontendPlugin.hpp:31,
                 from /work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0/src/compiler/AdaptiveCppClangPlugin.cpp:29:
/work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0/src/compiler/../../include/hipSYCL/compiler/Frontend.hpp: In function ‘std::string hipsycl::compiler::detail::buildKernelNameFromRecordType(const
 clang::QualType&, clang::MangleContext*)’:
/work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0/src/compiler/../../include/hipSYCL/compiler/Frontend.hpp:162:12: error: ‘class clang::MangleContext’ has no member named ‘mangleTypeName’; did yo
u mean ‘mangleName’?
  162 |   Mangler->mangleTypeName(RecordType, SS);
      |            ^~~~~~~~~~~~~~
      |            mangleName
```

**To Reproduce**
```
cmdline=cmake -D CMAKE_INSTALL_PREFIX=/work/00434/eijkhout/adaptivecpp/installation-adaptivecpp-23.10.0-milan-gcc13                     -D CMAKE_VERBOSE_MAKEFILE=ON                     -D BUILD_SHARED_LIBS=ON                     -D ACPP_VERSION_SUFFIX="-" -D BOOST_ROOT=/work/00434/eijkhout/boost/installation-boost-1.83.0-milan-gcc13 -D Boost_NO_SYSTEM_PATHS=On -DBoost_DEBUG=ON -D WITH_CUDA_BACKEND=OFF -DWITH_ROCM_BACKEND=OFF -DWITH_LEVEL_ZERO_BACKEND=OFF -D BUILD_CLANG_PLUGIN=OFF -D CMAKE_CXX_COMPILER=g++ -D WITH_ACCELERATED_CPU=OFF -D FIXES_FOR_MISSING_LIB_CLANG=ON -D WITH_SSCP_COMPILER=OFF -D WITH_OPENCL_BACKEND=OFF -D WITH_LEVEL_ZERO_BACKEND=OFF                      /work/00434/eijkhout/adaptivecpp/adaptivecpp-23.10.0/
```

Clang was found:
```
-- Building hipSYCL against LLVM configured from /work/00434/eijkhout/clang/installation-clang-git-milan-gcc13/lib/cmake/llvm
-- Selecting clang: /work/00434/eijkhout/clang/installation-clang-git-milan-gcc13/bin/clang++
-- Using clang include directory: /work/00434/eijkhout/clang/installation-clang-git-milan-gcc13/lib/clang/18/include/..
```

--- a/include/hipSYCL/compiler/Frontend.hpp
+++ b/include/hipSYCL/compiler/Frontend.hpp
@@ -159,7 +159,7 @@ class CompleteCallSet : public clang::RecursiveASTVisitor<CompleteCallSet> {
 inline std::string buildKernelNameFromRecordType(const clang::QualType &RecordType, clang::MangleContext *Mangler) {
   std::string KernelName;
   llvm::raw_string_ostream SS(KernelName);
-  Mangler->mangleTypeName(RecordType, SS);
+  Mangler->mangleCanonicalTypeName(RecordType, SS);
 
   return KernelName;
 }

--- a/include/hipSYCL/compiler/Frontend.hpp
+++ b/include/hipSYCL/compiler/Frontend.hpp
@@ -159,7 +159,11 @@ class CompleteCallSet : public clang::RecursiveASTVisitor<CompleteCallSet> {
 inline std::string buildKernelNameFromRecordType(const clang::QualType &RecordType, clang::MangleContext *Mangler) {
   std::string KernelName;
   llvm::raw_string_ostream SS(KernelName);
+#if LLVM_VERSION_MAJOR >= 18
   Mangler->mangleCanonicalTypeName(RecordType, SS);
+#else
+  Mangler->mangleTypeName(RecordType, SS);
+#endif
 
   return KernelName;
 }

